### ISSUE TITLE
[trigger-sim] Needs some attention (Trac #966)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#960]
Project docs suck and are horribly outdated.  The wiki docs need to be moved to project-level sphinx docs.

There are two I3TimeShifter modules.  There can be only one.  I'm leaning towards the python version because that's what we've been using in recent production.

The test coverage is not the worst I've seen, but it can be better.

Migrated from https://code.icecube.wisc.edu/ticket/966
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:11:45", 
    "description": "Project docs suck and are horribly outdated.  The wiki docs need to be moved to project-level sphinx docs.\n\nThere are two I3TimeShifter modules.  There can be only one.  I'm leaning towards the python version because that's what we've been using in recent production.\n\nThe test coverage is not the worst I've seen, but it can be better.", 
    "reporter": "olivas", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067105393059", 
    "component": "combo simulation", 
    "summary": "[trigger-sim] Needs some attention", 
    "priority": "blocker", 
    "keywords": "", 
    "time": "2015-05-07T02:42:02", 
    "milestone": "", 
    "owner": "olivas", 
    "type": "cleanup"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150519T21:16:37: olivas changed status from "new" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20150529T16:23:20: lraedel commented:

> Dataset 11065 includes events where the start of the global trigger is not at zero. In the processing chain this leads to errors in millipede because pulses are outside the readout window.

**Comment by icecube-trac** [comment]
Trac update at 20150529T16:47:07: olivas commented:

> Similar bugs have been reported in the past and this seems to have the same origin.
> 
> Some current-ish production uses the python version of the time-shifter, which shifts everything by type.  If you're using the C++ module you have to be very explicit about what you want shifted.  It's very easy to get time skews between frame objects, which I think is what we're seeing here.
> 
> This dataset (though current) uses an old version of simprod-scripts, which uses the C++ version of the time shifting module, and appears to be misconfigured for this dataset.
> 
> http://code.icecube.wisc.edu/svn/projects/simprod-scripts/releases/V01-05-09/python/simulation/detector.py
>

**Comment by icecube-trac** [comment]
Trac update at 20150721T11:47:52: david.altmann commented:

> I think this problem also occurs for dataset 11077.
>

**Comment by icecube-trac** [comment]
Trac update at 20150922T01:40:00: olivas changed milestone from "IceSim 4.2" to "IceSim 5.0"

**Comment by icecube-trac** [comment]
Trac update at 20151030T19:37:38: 
* olivas changed resolution from "" to "fixed"
* olivas changed status from "accepted" to "closed"
* olivas commented:

> In changeset 138999:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="138999"
> updated documentation.  trigger-sim is in pretty good shape.  this finally closes #960.  trigger-sim has gotten a *lot* of attention.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:11:45: 
* olivas changed milestone from "IceSim 5.0" to ""
* olivas commented:

> Milestone IceSim 5.0 deleted
